src/common/allowance-split-button/index.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import styled from 'styled-components';
+import SplitButton, { SplitButtonOptions } from 'common/split-button';
+import { FormattedMessage } from 'react-intl';
+import Tooltip from '@mui/material/Tooltip';
+import HelpOutlineIcon from '@mui/icons-material/HelpOutline';
+
+interface AllowanceSplitButtonProps {
+  onClick: () => void;
+  options: SplitButtonOptions;
+  text: React.ReactNode;
+  symbol: string;
+  disabled?: boolean;
+}
+
+const StyledTooltip = styled(Tooltip)`
+  margin-bottom: 2px;
+  margin-left: 5px;
+`;
+
+export const AllowanceTooltip = (props: { symbol: string }) => {
+  const { symbol } = props;
+  return (
+    <StyledTooltip
+      title={
+        <FormattedMessage
+          description="Allowance Tooltip"
+          defaultMessage="You must give the Mean Finance smart contracts permission to use your {symbol}"
+          values={{
+            symbol,
+          }}
+        />
+      }
+      arrow
+      placement="top"
+    >
+      <HelpOutlineIcon fontSize="small" />
+    </StyledTooltip>
+  );
+};
+
+const AllowanceSplitButton = (props: AllowanceSplitButtonProps) => {
+  const { onClick, options, text, disabled, symbol } = props;
```

Comment:
I would actually pass the token and the amount here, let the button define which text is going to show.

I would only put this props for the button

```
disabled?: boolean
onMaxApprove: () => void
onApproveExact: (amount: BigNumber) => void
token: Token
amount: BigNumber
```